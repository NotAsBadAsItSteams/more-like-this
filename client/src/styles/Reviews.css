:local .element {
  background-color: #677886;
  color: #C0D4E0;
  padding: 10px;
  font-size: 12px;
  font-family: Tahoma, Arial, Helvetica, sans-serif;
  border-radius: 2px;
  width: 285px;
}

:local .positive {
  color: #65c0f4;
}

:local .negative {
  color: #c35c2c;
}

:local .mixed {
  color: #a89269;
}

