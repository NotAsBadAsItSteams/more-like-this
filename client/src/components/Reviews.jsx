import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Reviews.css';

const ratingType = (rating) => {
  if (rating.toLowerCase().includes('positive')) {
    return 'positive';
  } else if (rating.toLowerCase().includes('negative')) {
    return 'negative';
  } else if (rating.toLowerCase().includes('mixed')) {
    return 'mixed';
  }
  return null;
};

const formatNumber = number => number.toLocaleString();

const Reviews = ({ number, rating }) => (
  <div className={styles.element}>
    <div>Overall user reviews: </div>
    <span className={styles[ratingType(rating)]}>{rating}</span>
    <span> ({formatNumber(number)} reviews)</span>
  </div>
);

Reviews.propTypes = {
  number: PropTypes.number.isRequired,
  rating: PropTypes.string.isRequired,
};

export default Reviews;
