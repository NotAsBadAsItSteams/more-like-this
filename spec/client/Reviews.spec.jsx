import React from 'react';
import renderer from 'react-test-renderer';
import Reviews from '../../client/src/components/Reviews';

describe('Reviews component', () => {
  test('should match snapshot for mostly negative review', () => {
    const reviewRating = 'Mostly Negative';
    const reviewNumber = 20007;

    const component = renderer.create(<Reviews rating={reviewRating} number={reviewNumber} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot for mostly positive review', () => {
    const reviewRating = 'Mostly Positive';
    const reviewNumber = 107;

    const component = renderer.create(<Reviews rating={reviewRating} number={reviewNumber} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match snapshot for mixed review', () => {
    const reviewRating = 'Mixed';
    const reviewNumber = 12817;

    const component = renderer.create(<Reviews rating={reviewRating} number={reviewNumber} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
