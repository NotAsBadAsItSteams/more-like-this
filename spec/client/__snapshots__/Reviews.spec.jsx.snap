// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component should match snapshot for mixed review 1`] = `
<div
  className="element"
>
  <div>
    Overall user reviews: 
  </div>
  <span
    className="mixed"
  >
    Mixed
  </span>
  <span>
     (
    12,817
     reviews)
  </span>
</div>
`;

exports[`Reviews component should match snapshot for mostly negative review 1`] = `
<div
  className="element"
>
  <div>
    Overall user reviews: 
  </div>
  <span
    className="negative"
  >
    Mostly Negative
  </span>
  <span>
     (
    20,007
     reviews)
  </span>
</div>
`;

exports[`Reviews component should match snapshot for mostly positive review 1`] = `
<div
  className="element"
>
  <div>
    Overall user reviews: 
  </div>
  <span
    className="positive"
  >
    Mostly Positive
  </span>
  <span>
     (
    107
     reviews)
  </span>
</div>
`;
